src/reconcile/rec_write.c

```diff
@@ -3229,6 +3229,18 @@ __rec_col_int(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
 	WT_RET(__rec_split_init(
 	    session, r, page, page->pg_intl_recno, btree->maxintlpage));
 
+	/*
+	 * We need to mark this page as splitting, as this may be an in-memory
+	 * split during a checkpoint.
+	 */
+	for (;;) {
+		F_CAS_ATOMIC(page, WT_PAGE_SPLITTING, ret);
```

Comment:
We talked about renaming this flag now that it is used to prevent splits during reconciliation as well as to coordinate between multiple threads wanting to split child pages.

How about `WT_PAGE_SPLIT_LOCKED`?